chat-plugins/datasearch.js

```diff
@@ -719,7 +719,8 @@ function runDexsearch(target, cmd, canAll, message) {
 		}
 	}
 	let results = [];
-	for (let mon in dex) {
+	let monNames = Object.keys(dex).sort();
```

Comment:
i dont really see a reason for this to be defined independently when the only time you use it is for the ``for...of``` loop

Reply:
if I just do `for (const mon of Object.keys(dex).sort())` then I think it'll rerun the sort() for every iteration in then loop? though I'm not super familiar with the mechanics of that in js.

Reply:
No, it only evaluates it once.

    > for (const x of [3, 2, 1].sort((a, b) => { console.log(a, b); return a - b; })) console.log(x)
    3 2
    3 1
    2 1
    1
    2
    3

Reply:
Ah great, thanks!